src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -2731,8 +2731,24 @@ void CodeGen::genCodeForReturnTrap(GenTreeOp* tree)
     if (helperFunction.accessType == IAT_VALUE)
     {
         // If the helper is a value, we need to use the address of the helper.
-        params.addr     = helperFunction.addr;
-        params.callType = EC_FUNC_TOKEN;
+        params.addr = helperFunction.addr;
+        if (GetEmitter()->IsAddressInRange(params.addr))
```

Comment:
Maybe we should dedup these blocks via a helper: https://www.diffchecker.com/zfB1Id3J/